### DISCUSSION TITLE
Unit test with multiple operators is it possible?

### DISCUSSION THREAD

**Discussion by georgen117** [microsoft/onnxruntime#11076] [Other Q&A] [answered]
Is there a way to write a unit test that takes multiple operators.

For example if I want to test the output of `MatMul` -> `Add` -> `Relu` 

is there a way to specify a graph and the 3 inputs needed to satisfy the graph as well as the expected output in a unit test?

The closest I have seen to this is in the optimizer/graph_transformer_test.cc but it optimizes the graph never checks the actual output.

**Comment by hariharans29** [comment, marked as answer]
By inheriting from `OpTester` and overriding `AddNodes()` - like [this](https://github.com/microsoft/onnxruntime/blob/eab7c0d5bfbe69895bfca6715d0aee216d0ed0a9/onnxruntime/test/providers/cpu/controlflow/if_test.cc#L45) ?

**Reply by georgen117** [reply]
@hariharans29 thanks so much for pointing me directly to the example code it was really helpful.
